src/nodes.coffee

```diff
@@ -1894,6 +1897,66 @@ exports.Obj = class Obj extends Base
       answer.push @makeCode join
     if @front then @wrapInParentheses answer else answer
 
+  # Convert "bare" properties to `ObjectProperty`s (or `Splat`s)
+  expandProperty: (property) ->
+    {variable, context, operatorToken} = property
+    key = if property instanceof Assign and context is 'object'
+      variable
+    else if property instanceof Assign
+      operatorToken.error "unexpected #{operatorToken.value}" unless @lhs
+      variable
+    else
+      property
+    if key instanceof Value and key.hasProperties()
+      key.error 'invalid object key' unless context isnt 'object' and key.this
+      if property instanceof Assign
+        return new ObjectProperty fromAssign: property
+      else
+        return new ObjectProperty key: property
+    return new ObjectProperty(fromAssign: property) unless key is property
+    return property if property instanceof Splat
+
+    new ObjectProperty key: property
+
+  expandProperties: ->
+    @expandProperty(property) for property in @properties
+
+  astType: -> 'ObjectExpression'
+
+  astProperties: ->
+    properties:
+      property.ast() for property in @expandProperties()
```

Comment:
So now to generate `properties` on an `Obj`'s AST we can just delegate to children's `.ast()` methods in the typical way